easy-random-protobuf/src/test/java/org/jeasy/random/protobuf/Protobuf3MessageGenerationTest.java

```diff
@@ -210,4 +210,15 @@ void shouldUseCustomRandomizerForSubMessageWhenItsDefined() {
         assertThat(protoInstance.hasEmbeddedMessage()).isTrue();
         assertThat(protoInstance.getEmbeddedMessage()).isEqualTo(customEmbeddedMessage);
     }
+
+    @Test
+    void shouldUseCustomProviderRegistryToFillFields(){
```

Comment:
IMO we need to add support for byType as well.

Reply:
will take a look, thanks for the feedback!